Detectors/TRD/workflow/src/TRDTrackletTransformerSpec.cxx

```diff
@@ -0,0 +1,92 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "TRDWorkflow/TRDTrackletTransformerSpec.h"
+
+#include "DataFormatsTRD/TriggerRecord.h"
+#include "DataFormatsTRD/Tracklet64.h"
+
+#include "DataFormatsTRD/CalibratedTracklet.h"
+
+using namespace o2::framework;
+using SubSpecificationType = o2::framework::DataAllocator::SubSpecificationType;
+
+namespace o2
+{
+namespace trd
+{
+
+void TRDTrackletTransformerSpec::init(o2::framework::InitContext& ic)
+{
+  LOG(info) << "initializing tracklet transformer";
+}
+
+void TRDTrackletTransformerSpec::run(o2::framework::ProcessingContext& pc)
+{
+  LOG(info) << "running tracklet transformer";
+
+  std::vector<CalibratedTracklet> calibratedTracklets;
+  std::vector<Tracklet64> tracklets = pc.inputs().get<std::vector<Tracklet64>>("inTracklets");
+  // std::vector<TriggerRecord> triggerRec = pc.inputs().get<std::vector<TriggerRecord>>("triggerRecord");
+
+  // for (int reci=0; reci < triggerRec.size(); reci++)
+  // {
+  //   LOG(info) << triggerRec[reci].getFirstEntry() << " | " << triggerRec[reci].getNumberOfObjects();
+  // }
+
+  LOG(info) << tracklets.size() << " tracklets found!";
+
+  for (int itracklet; itracklet < tracklets.size(); itracklet++) {
+
+    Tracklet64* tracklet = &tracklets[itracklet];
+
+    uint64_t hcid = tracklet->getHCID();
+    uint64_t padrow = tracklet->getPadRow();
+    uint64_t column = tracklet->getColumn();
+    // 0-2048 | units:pad-widths | granularity=1/75 (measured from center pad 10) 1024 is 0/cetner of pad 10
+    uint64_t position = tracklet->getPosition();
+    // 0-127 | units:pads/timebin | granularity=1/1000
+    uint64_t slope = tracklet->getSlope();
+
+    std::vector<double> spacePoint = mTransformer.transformToSpacePoint(hcid, padrow, column, position);
+
+    double rawDy = mTransformer.calculateDy(hcid, slope);
+
+    float t0Correction = -0.1;
+    float oldLorentzAngle = 0.16;
+    float lorentzAngle = -0.14;
+    float driftVRatio = 1.1;
```

Comment:
Yeah, those will definitely disappear later, just not totally sure where to at the moment. I would think that my `TrackletTransformer` will pull them from the CCDB. I added a comment explaining that they are only temporary.

Reply:
In the end, these will have to come out of the CCDB. I think the TrackletTransformer would be the best place to look up these values for the chamber in question, and do all the conversion internally: i.e. you pass it a Tracklet64, and ideally, you get a CalibratedTracklet back.

I think the code will get quite messy when we put the whole CCDB code into the DPL device. 

I'm not sure how this can be combined with an emplace_back().

Reply:
Hi Tom. Yes, okay the CCDB business will happen in `TrackletTransformer`. 

Are you also saying that you would like `TrackletTransformer` to include a function able to convert `Tracklet64` into `CalibratedTracklet`? So that this can be done without using the workflow?